src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops.cpp

```diff
@@ -294,41 +299,73 @@ PerformanceImplicitGemmWrwV4R4Xdlops::CalculateGridSize(const ConvolutionContext
     return std::make_tuple(GridSize, true);
 }
 
-int PerformanceImplicitGemmWrwV4R4Xdlops::CalculateGemmKBlocks(const ConvolutionContext& ctx) const
+std::tuple<int, int, int, int, int, bool>
+PerformanceImplicitGemmWrwV4R4Xdlops::CalculateGemmSizeAndGemmKBlock(
+    const ConvolutionContext& ctx) const
 {
-    const int MaxBlocks = 2400;
-    int GemmKBlocks     = 1;
-    int gemm_g          = -1;
-    int gemm_m          = -1;
-    int gemm_n          = -1;
-    int gemm_k_total    = -1;
-
-    std::tie(gemm_g, gemm_m, gemm_n, gemm_k_total) =
-        ConvHipImplicitGemmWrwV4R4Xdlops::CalculateGemmSize(ctx);
-    const auto n = ConvolutionContextInterpreter::GetBatchN(ctx);
-    if(!(gemm_m % GemmMPerBlock == 0 && gemm_n % GemmNPerBlock == 0))
-        return GemmKBlocks;
-    int GridSize = gemm_g * (gemm_m / GemmMPerBlock) * (gemm_n / GemmNPerBlock);
-    int Blocks   = std::max(MaxBlocks / GridSize, 1);
-    Blocks       = std::min(Blocks, n);
-
-    if(gemm_k_total % GemmKPack != 0)
-        return GemmKBlocks;
-    const auto gemm_k = gemm_k_total / GemmKPack;
-    for(; Blocks > 1; Blocks--)
+    int gemm_g       = -1;
+    int gemm_m       = -1;
+    int gemm_n       = -1;
+    int gemm_k_total = -1;
+    int gemm_k_block = -1;
+
+    try
     {
-        if(n % Blocks != 0)
-            continue;
-        if(gemm_k % Blocks != 0)
-            continue;
-        const auto gemm_k_sub = gemm_k / Blocks;
-        if(!(gemm_k_sub % GemmKPerBlock == 0))
-            continue;
-        break;
+        const auto g  = ConvolutionContextInterpreter::GetGroupCountG(ctx);
+        const auto n  = ConvolutionContextInterpreter::GetBatchN(ctx);
+        const auto k  = ConvolutionContextInterpreter::GetOutputChannelK(ctx);
+        const auto c  = ConvolutionContextInterpreter::GetInputChannelC(ctx);
+        const auto ho = ConvolutionContextInterpreter::GetOutputHeightHo(ctx);
+        const auto wo = ConvolutionContextInterpreter::GetOutputWidthWo(ctx);
+        const auto y  = ConvolutionContextInterpreter::GetFilterHeightY(ctx);
+        const auto x  = ConvolutionContextInterpreter::GetFilterWidthX(ctx);
+
+        const auto k_per_group = k / g;
+        const auto c_per_group = c / g;
+
+        gemm_m = k_per_group;
+        gemm_n = c_per_group * y * x;
+
+        int gemm_k_block_times_gemm_k_total = n * ho * wo;
```

Comment:
Suggestion: For more ituitive naming, gemm_k_total = n * ho * wo. But gemm_k_per_g = gemm_k_total / gemm_k_block.

Reply:
Will fix the naming for GEMM uniformly in future PR